openmdao_book/features/debugging/profiling/inst_call_tracing.ipynb

```diff
@@ -0,0 +1,279 @@
+    "                    PETScKrylov#0.PETScKrylov._declare_options\n",
+    "                    <-- PETScKrylov#0.PETScKrylov._declare_options (time:  0.06384) (total: 75.445 MB)\n",
+    "                <-- PETScKrylov#0.Solver.__init__ (time:  0.06391) (total: 75.445 MB)\n",
+    "            <-- PETScKrylov#0.LinearSolver.__init__ (time:  0.06397) (total: 75.445 MB)\n",
+    "        <-- PETScKrylov#0.PETScKrylov.__init__ (time:  0.06402) (total: 75.445 MB)\n",
+    "    <-- DistribOuptutImplicit#0(aero.icomp).System._setup_procs (time:  0.06519) (total: 77.371 MB) (diff: +4772 KB)\n",
+    "    DistribInputExplicit#0().System._setup_procs\n",
+    "    <-- DistribInputExplicit#0(aero.ecomp).System._setup_procs (time:  0.06738) (total: 79.281 MB) (diff: +1956 KB)\n",
+    "<-- Group#1(aero).Group._setup_procs (time:  0.06746) (total: 79.281 MB)\n",
+    "\n",
+    "...\n",
+    "```\n",
+    "\n",
+    "Note that total memory usage and elapsed time is shown on each function return line. Those function returns where a difference in memory was found will display the difference at the end of the line.\n",
+    "\n",
+    "The tracer can also be used to help track down memory leaks. Using the -*l* option, it will display a list of object types and their counts that were created since the function was called and not garbage collected after the function returned. For example:\n",
+    "\n",
+    "``` bash\n",
+    "    openmdao trace -l <your_python_script_here> -g solver\n",
+    "```\n",
+    "\n",
+    "will result in output like this:\n",
+    "\n",
+    "```\n",
+    "...\n",
+    "\n",
+    "LinearRunOnce#0.LinearRunOnce.solve\n",
+    "    LinearRunOnce#0.LinearBlockGS._iter_execute\n",
+    "        LinearRunOnce#1.LinearRunOnce.solve\n",
+    "            LinearRunOnce#1.LinearBlockGS._iter_execute\n",
+    "                PETScKrylov#0.PETScKrylov.solve\n",
+    "                <-- PETScKrylov#0.PETScKrylov.solve\n",
+    "            <-- LinearRunOnce#1.LinearBlockGS._iter_execute\n",
+    "        <-- LinearRunOnce#1.LinearRunOnce.solve\n",
+    "           Recording +1\n",
+    "    <-- LinearRunOnce#0.LinearBlockGS._iter_execute\n",
+    "<-- LinearRunOnce#0.LinearRunOnce.solve\n",
+    "   Recording +1\n",
+    "\n",
+    "...\n",
+    "```\n",
+    "\n",
+    "This output shows that in LinearRunOnce\\#1.LinearRunOnce.solve, a Recording object was created and not garbage collected. Note that this does not always indicate a memory leak, as there are some functions that intentionally create new objects that are intended to last beyond the life of the function. This tool merely gives you a place to look in the code where a memory leak *might* exist.\n",
+    "\n",
+    "To see a list of the available pre-defined sets of functions to trace, look at the usage info for the -*g* command that can be obtained as follows:\n",
+    "\n",
+    "``` bash\n",
+    "    openmdao trace -h\n",
```

Comment:
I think it would work if we put !openmdao trace -h in a code block if we don't want to hard code the output.

Reply:
Done